packages/react-components/src/components/gauge/gaugeBase.tsx

```diff
@@ -0,0 +1,184 @@
+// eslint-disable-next-line import/default
+import React, { useCallback, useEffect } from 'react';
+import {
+  DEFAULT_GAUGE_PROGRESS_COLOR,
+  DEFAULT_GAUGE_SETTINGS,
+  DEFAULT_GAUGE_SETTINGS_WITH_THRESHOLDS,
+} from './constants';
+import { useECharts, useLoadableEChart } from '../../hooks/useECharts';
+import { GaugeBaseProperties } from './types';
+import { StyledThreshold, ThresholdValue } from '@iot-app-kit/core';
+import { getPreciseValue } from '../../utils/getPreciseValue';
+
+const getDecimalValue = ({
+  value,
+  min = 0,
+  max = 100,
+}: {
+  value: ThresholdValue;
+  min?: number;
+  max?: number;
+}) => {
+  if (typeof value !== 'number') return 0;
+  if (value <= min) return 0;
+  if (value >= max) return 1;
+  return (value - min) / (max - min);
+};
+
+export const GaugeBase: React.FC<GaugeBaseProperties> = ({
+  propertyPoint,
+  thresolds = [],
+  settings,
+  unit,
+  name,
+  isLoading,
+  significantDigits,
+  ...options
+}) => {
+  // Setup instance of echarts
+  const { ref, chartRef } = useECharts(options?.theme);
+
+  const getThresholds = (thresholds: StyledThreshold[]) => {
+    const newArray = thresholds.map(({ value, color }) => {
+      return [
+        getDecimalValue({ value, min: settings?.yMin, max: settings?.yMax }),
+        color,
+      ];
+    });
+
+    /*
+     * Add default threshold at 100% if none is defined
+     * This is required for ECharts to show the gauge correctly
+     */
+    if (!newArray.some((item) => item[0] === 1)) {
+      newArray.push([1, DEFAULT_GAUGE_PROGRESS_COLOR]);
+    }
+
+    return newArray;
+  };
+
+  const value = propertyPoint?.y;
+  const hasThresholds = thresolds.length > 0;
+  const gaugeThresholds = hasThresholds ? getThresholds(thresolds) : null;
+
+  useEffect(() => {
+    const gauge = chartRef.current;
+    // Set default Gauge options
+    gauge?.setOption(
+      !hasThresholds
+        ? DEFAULT_GAUGE_SETTINGS
+        : DEFAULT_GAUGE_SETTINGS_WITH_THRESHOLDS
+    );
+  }, [chartRef, hasThresholds]);
+
+  const getFormatterValue = useCallback(
+    (value: number) => {
+      if (!value) return '-';
+      return `{value|${getPreciseValue(value, significantDigits)}} ${
+        settings?.showUnit && unit ? '{unit| ' + unit + '}' : ''
+      }`;
+    },
+    [settings?.showUnit, significantDigits, unit]
+  );
+
+  useEffect(() => {
```

Comment:
this useEffect is quite long and handles a lot of logic
please move all this logic into a new folder/file under `packages/react-components/src/components/gauge/`

and break up the logic into helper functions/folders depending on what each section controls
example: move lines 131-164 which handles thresholds into a new file under `packages/react-components/src/components/gauge/thresholds`